### PR TITLE
Added sippreg_unregister API function

### PR DESCRIPTION
**Pull request by juha-h**

### CODE REVIEW

File: src/sipreg/reg.c [hunk 1]
```diff
@@ -450,6 +453,21 @@ int sipreg_register(struct sipreg **regp, struct sip *sip, const char *reg_uri,
...
 
+/**
+ * Unregisters SIP Registration client
+ *
+ * @param reg   SIP Registration client
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int sipreg_unregister(struct sipreg *reg)
+{
```
Comment (alfredh) [inline src/sipreg/reg.c:464]: if `reg` is NULL then return EINVAL

**Comment by juha-h** [comment]
Regarding `response_handler` change, would this look cleaner?
```
 out:
	if (!reg->expires) {
		if (msg && msg->scode >= 400 && msg->scode < 500)
			reg->fbregint = 0;

		if (!reg->terminated) {
			if (reg->fbregint)
				tmr_start(&reg->tmr, reg->fbregint * 1000,
					  tmr_handler, reg);
			reg->resph(err, msg, reg->arg);
		}
		else {
			mem_deref(reg);
		}
	}
```

**Comment by cspiel1** [comment]
One if level less by:

```c
	if (!reg->expires) {
		if (msg && msg->scode >= 400 && msg->scode < 500)
			reg->fbregint = 0;

		if (reg->terminated) {
			mem_deref(reg);
		}
		else if (reg->fbregint) {
			tmr_start(&reg->tmr, reg->fbregint * 1000, tmr_handler,
				  reg);
			reg->resph(err, msg, reg->arg);
		}
		else {
			reg->resph(err, msg, reg->arg);
		}
}
```

**Comment by juha-h** [comment]
Yours has more code then mine (reg->resph(err, msg, reg->arg); repeated twice).

**Comment by sreimers** [comment]
Maybe:
```c
	if (!reg->expires) {
		if (msg && msg->scode >= 400 && msg->scode < 500)
			reg->fbregint = 0;

		if (reg->terminated) {
			mem_deref(reg);
			return;
		}

		if (reg->fbregint)
			tmr_start(&reg->tmr, reg->fbregint * 1000,
					  tmr_handler, reg);

		reg->resph(err, msg, reg->arg);

	}
```

**Comment by juha-h** [comment]
Is there any objections left for merging this commit?

**Comment by alfredh** [comment]
please go ahead :)

**Pull request merged**